service.go

```diff
@@ -308,14 +346,15 @@ func (s *Service) registerValidatorForClient(_ctx context.Context, req *relaygrp
 	})
 }
 func (s *Service) StartStreamHeaders(ctx context.Context, wg *sync.WaitGroup) {
-
-	for _, client := range s.streamingClients {
+	s.dialerClients.mu.RLock()
+	for _, client := range s.dialerClients.streamingClients {
 		wg.Add(1)
 		go func(_ctx context.Context, c *common.Client) {
 			defer wg.Done()
 			s.handleStream(_ctx, c)
```

Comment:
in these stream functions, the _ctx is replaced with `	ctx = trace.ContextWithSpan(context.Background(), parentSpan)
` 
will the cancel function work?